app/pages/friends/hooks/useUsers.ts

```diff
@@ -0,0 +1,59 @@
+import { useQuery } from "@tanstack/react-query";
+import { useAuth } from "@/pages/friends/hooks/useAuth";
+
+interface User {
+  id: number;
+  name: string;
+  email: string;
+  imageUrl: string | null;
+  dateJoined: string;
+}
+
+export function useUsers(userIds: number[]) {
+  const { getAuthHeader } = useAuth();
+
+  const { data: users, isLoading } = useQuery({
+    queryKey: ["users", userIds],
+    queryFn: async () => {
+      if (userIds.length === 0) return [];
+
+      const userResults: User[] = [];
+
+      for (const id of userIds) {
+        try {
+          const response = await fetch(
+            `http://localhost:8080/api/v1/users/${id}/profile`,
+            {
+              headers: getAuthHeader(),
+            }
+          );
+
+          const user = await response.json();
+
+          userResults.push({
+            id: user.id,
+            name: user.name,
+            email: user.email,
+            imageUrl: user.imageUrl,
+            dateJoined: user.dateJoined
+              ? new Date(user.dateJoined).toISOString()
+              : "",
+          });
+        } catch (error) {
+          console.error(
+            `Błąd przy pobieraniu danych użytkownika o ID: ${id}`,
+            error
+          );
+        }
+      }
+
```

Comment:
very nice that you have understaning of how to implement fetching logic with auth headers . cool one 

But

we have already implemented it : ) look for other code parts before trying to implement your own one. 

we dont need all those types you have declared since they are autogerated . we also dont need to use fetch library and wrtite paths for queries because we are using autogenerated functions 


also React query (useQuery hook) caches data under the hood by queryKey, so when you are fetching [ "users", ["1","2","42"] ] it will  cache data for 3 users of ids 1 2 and 42. ok . but if list will change - for example  we will need new user [ "users", ["1","2","42", "92"] ] then it will refetch 1 2 42 again and user 92 too - we dont want this.  Instead we want to cache user 1 2 42 92 independantly so if list changes then we will fetch only 92. This can be achieved by multiple useQuery hooks in one hook. 

I understand that this is overwhelming so i will imelemnt this for you and show all together